python/test/unit/runtime/test_compilation_listener.py

```diff
@@ -0,0 +1,90 @@
+import triton
+import triton.language as tl
+
+from triton.backends.compiler import GPUTarget
+from triton.compiler.compiler import ASTSource, IRSource
+
+from triton.runtime.config import TritonConfig, CompileTimes
+from typing import Any, Union
+
+import torch
+import uuid
+
+
+def get_add_kernel(suffix: str) -> triton.JITFunction:
+
+    def add_kernel(
+        in_ptr0,
+        in_ptr1,
+        out_ptr,
+        n_elements,
+        BLOCK_SIZE: "tl.constexpr",
+    ):
+        pid = tl.program_id(axis=0)
+        block_start = pid * BLOCK_SIZE
+        offsets = block_start + tl.arange(0, BLOCK_SIZE)
+        mask = offsets < n_elements
+        x = tl.load(in_ptr0 + offsets, mask=mask)
+        y = tl.load(in_ptr1 + offsets, mask=mask)
+        output = x + y
+        tl.store(out_ptr + offsets, output, mask=mask)
+
+    add_kernel.__name__ += suffix
+
+    return triton.jit(add_kernel)
+
+
+def test_compile_stats(device: str) -> None:
+    captured: tuple[Union[ASTSource, IRSource], dict[str, Any], CompileTimes, bool] | None = None
+
+    # We generate a unique kernel name so that we don't pick up cache hit on
+    # first run.
+    suffix = str(uuid.uuid4()).replace("-", "")
```

Comment:
NIT: You could use the `fresh_triton_cache` pytest fixture, this would ensure there's never a cache hit without needing the kernel rename trick.

Reply:
Nice! I'm glad there was a better way than my hack, now to go clean up `add_kernel*` in my cache...